hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/StateContext.java

```diff
@@ -73,6 +73,7 @@
   private final Queue<ContainerAction> containerActions;
   private final Queue<PipelineAction> pipelineActions;
   private DatanodeStateMachine.DatanodeStates state;
+  private boolean shutdownStateMachine = false;
```

Comment:
Can we avoid this extra flag? Instead can `getShutdownStateMachine` just check whether `this.state == DatanodeStateMachine.DatanodeStates.SHUTDOWN`

Reply:
This additional check is added to do shutdown only when the state is changed to shutdown state by executing any of the datanode state tasks.